definitions/npm/react-i18next_v6.x.x/test_react-i18next_v6.x.x.js

```diff
@@ -0,0 +1,175 @@
+// @flow
+
+import * as React from "react";
+import {
+  loadNamespaces,
+  translate,
+  I18nextProvider,
+  I18n,
+  Interpolate,
+  Trans,
+  setDefaults,
+  reactI18nextModule,
+  defaultOptions,
+  getDefaults,
+  getI18n,
+  setI18n
+} from "react-i18next";
+import type { TFunction, Translator } from "react-i18next";
+
+const i18n = { loadNamespaces: () => {} };
+
+<I18nextProvider i18n={i18n} children={<div />} />;
+
+// $ExpectError - missing children prop
+<I18nextProvider i18n={i18n} />;
+// $ExpectError - missing i18n prop
+<I18nextProvider children={<div />} />;
+
+// passing
+loadNamespaces({ components: [], i18n });
+loadNamespaces({ components: [() => <div />], i18n });
+
+// $ExpectError - too few arguments
+loadNamespaces();
+// $ExpectError - wrong type
+loadNamespaces("");
+// $ExpectError - wrong component type
+loadNamespaces({ components: [{}], i18n });
+
+type OwnProps = { s: string };
+type Props = OwnProps & { t: TFunction };
+
+const Comp = ({ s, t }: Props) => <div prop1={t("")} prop2={" " + s} />;
+
+class ClassComp extends React.Component<Props> {
+  render() {
+    const { s, t } = this.props;
+    return <div prop={t("")} />;
+  }
+}
+
+// $ExpectError - wrong argument type
+const FlowErrorComp = ({ s, t }: Props) =>
+  <div
+    prop1={t("", "")} // misuse of t()
+    prop2={" " + s}
+  />;
+
+class FlowErrorClassComp extends React.Component<Props> {
+  render() {
+    // $ExpectError - wrong argument type
+    const { s, t } = this.props;
+    return <div prop={t({})} />; // misuse of t()
+  }
+}
+
+// $ExpectError - too few arguments
+translate();
```

Comment:
why is that an error, we can have `translate()` which means we are using defaultNS ==> https://react.i18next.com/components/translate-hoc.html